test/framework/src/main/java/org/opensearch/test/ConcurrentSearchIntegTestCase.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.junit.Before;
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.opensearch.common.settings.FeatureFlagSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
+
+public class ConcurrentSearchIntegTestCase extends OpenSearchIntegTestCase {
+
+    protected final TestParameters testParameters;
+    @ParametersFactory
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(
+            new Object[] {new TestParameters(true)},
+            new Object[] {new TestParameters(false)}
+        );
+    }
+    public ConcurrentSearchIntegTestCase(TestParameters testParameters) {
+        this.testParameters = testParameters;
+    }
+
+    @Before
+    void beforeTests() {
+        ClusterUpdateSettingsResponse clusterUpdateSettingsResponse = client().admin()
+            .cluster()
+            .prepareUpdateSettings()
+            .setPersistentSettings(Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), testParameters.isConcurrentSearchSettingVal())).get();
+        assertEquals(String.valueOf(testParameters.isConcurrentSearchSettingVal()), clusterUpdateSettingsResponse.getPersistentSettings().get(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey()));
+    }
+
+    // TODO: remove the function once the concurrent search experimental flag is removed
+    @Override
+    protected Settings featureFlagSettings() {
+        Settings.Builder featureSettings = Settings.builder();
+        for (Setting builtInFlag : FeatureFlagSettings.BUILT_IN_FEATURE_FLAGS) {
+            featureSettings.put(builtInFlag.getKey(), builtInFlag.getDefaultRaw(Settings.EMPTY));
+        }
+        featureSettings.put(FeatureFlags.CONCURRENT_SEGMENT_SEARCH, true);
+        return featureSettings.build();
+    }
+
+    public static class TestParameters {
```

Comment:
Those are `Settings`, could use them instead?

```
  @ParametersFactory
    public static Collection<Object[]> parameters() {
        return Arrays.asList(
            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), true},
            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), false}
        );
    }
```
?

Reply:
@reta yes that can be done, i thought about it but that would make parameterization limited to the Settings, I tried to make it such that it is not just Settings. Thoughts?

Reply:
Thanks @neetikasinghal , as it stays, the `TestParameters` are very generic and in fact, difficult to figure out what exactly is parameterized, whereas with `Settings` - it is very clear.